src/rtFileDownloader.cpp

```diff
@@ -676,6 +724,412 @@ void rtFileDownloader::clearFileCache()
+    }
+    if (downloadRequest->cors() != NULL)
+      downloadRequest->cors()->updateRequestForAccessControl(&list);
+    if (readDataSize > 0)
+    {
+      list = curl_slist_append(list, "Expect:");
+    }
+    curl_easy_setopt(curl_handle, CURLOPT_HTTPHEADER, list);
+    //CA certificates
+    // !CLF: Use system CA Cert rather than CA_CERTIFICATE fo now.  Revisit!
+    //curl_easy_setopt(curl_handle,CURLOPT_CAINFO,mCaCertFile.cString());
+    curl_easy_setopt(curl_handle, CURLOPT_SSL_VERIFYHOST, 2);
+    curl_easy_setopt(curl_handle, CURLOPT_SSL_VERIFYPEER, true);
+
+    /* some servers don't like requests that are made without a user-agent
+     field, so we provide one */
+    curl_easy_setopt(curl_handle, CURLOPT_USERAGENT, "libcurl-agent/1.0");
+
+    if (useProxy)
+
+    {
+        curl_easy_setopt(curl_handle, CURLOPT_PROXY, proxyServer.cString());
+        curl_easy_setopt(curl_handle, CURLOPT_PROXYTYPE, CURLPROXY_HTTP);
+    }
+    else
+    {
+      curl_easy_setopt(curl_handle, CURLOPT_PROXY, "");
+    }
+
+    if (true == headerOnly)
+    {
+      curl_easy_setopt(curl_handle, CURLOPT_NOBODY, 1);
+    }
+
+    if (!method.isEmpty() && method.compare("GET") != 0)
+    {
+      if (method.compare("POST") == 0)
+        curl_easy_setopt(curl_handle, CURLOPT_POST, 1L);
+      else if (method.compare("PUT") == 0)
+        curl_easy_setopt(curl_handle, CURLOPT_UPLOAD, 1L);
+      else
+        curl_easy_setopt(curl_handle, CURLOPT_CUSTOMREQUEST, method.cString());
+    }
+
+    if (readDataSize > 0)
+    {
+      chunk.downloadRequest = downloadRequest;
+      curl_easy_setopt(curl_handle, CURLOPT_READFUNCTION, ReadMemoryCallback);
+      curl_easy_setopt(curl_handle, CURLOPT_READDATA, (void *)&chunk);
+      curl_easy_setopt(curl_handle, CURLOPT_POSTFIELDSIZE, readDataSize);
+    }
+
+    for(int iLoop=0; iLoop<= multipleIntervals; iLoop++)
+    {
+      if(iLoop < multipleIntervals)
+      {
+        byteRange = std::to_string(startRange) + "-" + std::to_string(startRange + downloadRequest->byteRangeIntervals() -1);
+      }
+      else if(iLoop == multipleIntervals)
+      {
+        if(fileSize % downloadRequest->byteRangeIntervals())
+          byteRange = std::to_string(startRange) + "-" + std::to_string(fileSize-1);
+      }
+      curl_easy_setopt(curl_handle, CURLOPT_RANGE, byteRange.c_str());
+
+      res = curl_easy_perform(curl_handle);
+      if(res != CURLE_OK)
+        break;
+
+      if(iLoop == 0)
+      {
+        curl_easy_setopt(curl_handle, CURLOPT_HEADERFUNCTION, NULL);
+        curl_easy_setopt(curl_handle, CURLOPT_HEADERDATA, NULL);
+
+        // Parsing total filesize from Content-Range.
+        std::string httpHeaderStr(reinterpret_cast< char const* >((char *)chunk.headerBuffer));
+        size_t findContentRange = httpHeaderStr.find("Content-Range:");
+        if(findContentRange != std::string::npos)
+        {
+          std::string strContentRange = httpHeaderStr.substr(findContentRange+1);
```

Comment:
Please remove std::string usage. Please use rtString instead.

Reply:
Changed to rtString and pushed again.